src/components/Card.js

```diff
@@ -1,40 +1,85 @@
 //"Components" has been renamed to "components in github"
 //my computer throws an error when I import in "index.js"
 export default class Card {
-  constructor(data, cardSelector, handleImageClick) {
-    this.name = data.name;
-    this.link = data.link;
+  constructor(
+    { isLiked, name, link, _id },
+    cardSelector,
+    handleImageClick,
+    openDeleteForm,
+    addLikeButton,
+    removeLikeButton
+  ) {
+    this.name = name;
+    this.link = link;
+    this._isLiked = isLiked;
+    this.id = _id;
     this.cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
+    this._addLikeButton = addLikeButton;
+    this._removeLikeButton = removeLikeButton;
+    this.openDeleteForm = openDeleteForm;
+    this._popupDeleteForm = document.querySelector("#delete-card-modal");
   }
 
   _setEventListeners() {
+    // if (this._isLiked == true) {
+    //   this._addLikeButtonElement();
+    // } else {
+    //   this._removeLikeButtonElement();
+    // }
     //.card__like-button
     this.likeButton.addEventListener("click", () => {
-      this._handleLikeButton();
+      console.log("the like button has been pressed");
+      console.log(this._isLiked);
+      if (this._isLiked == false) {
+        console.log("the person has liked this post");
+        this._addLikeButton(this);
+        this._addLikeButtonElement();
```

Comment:
You need to **toggle likes** only in block `then` after a successful response from the server because if there is an error the user will not know about that and will see the toggled heart color.

Reply:
Now the like button is only toggled after a response from the Api